File: 950C.cpp
// http://codeforces.com/contest/950/problem/C

#include <bits/stdc++.h>

using namespace std;

const int N = 2e5 + 1;

char s[N];
vector <int> v[N];

int main() {
  scanf("%s", s);
  int n = strlen(s);
  int idx = 0, sz = 0;
  for (int i = 0; i < n; ++i) {
    if (s[i] == '0') {
      v[idx++].emplace_back(i + 1);
    } else {
      if (idx == 0) {
        puts("-1");
        return 0;
      }
      v[--idx].emplace_back(i + 1);
    }
    sz = max(sz, idx);
  }
  if (sz != idx) {
    return 0 * puts("-1");
  }
  printf("%d\n", idx);
  for (int i = 0; i < idx; ++i) {
    printf("%d", v[i].size());
    for (int x : v[i]) {
      printf(" %d", x);
    }
    puts("");
  } 
  return 0;
}

File: solutions/950C.py
# http://codeforces.com/problemset/problem/950/C

s = input()
v = []
for x in range (200000) :
  v.append([])
idx = 0
pos = 0
sz = 0
for x in s : 
  if (x == '0') :
    v[idx].append(pos + 1)
    idx += 1
  else :
    if (idx == 0) :
      print("-1")
      exit()
    idx -= 1
    v[idx].append(pos + 1)
  pos += 1
  sz = max(sz, idx)
if (sz != idx) : 
  print("-1")
  exit()
print (idx)
at = 0
while (at < idx) : 
  print (len(v[at]), end = " ")
  att = 0
  for x in v[at] :
    print (x, end = " ")
  at += 1
  print("")